### PR TITLE
Redo the "stop explicitly requiring `--environment=dev` when running pixi install tasks" with new pixi release

### LINKED ISSUE #1324: Stop explicitly requiring `--environment=dev` when running pixi install tasks
When installing packages with `pixi run install`, Python packages are installed in the `default` environment instead of `dev`.
Because of that, we recommend running `pixi run --environment=dev install` at the moment.
https://github.com/Deltares/Ribasim/blob/0c02f9a356fd7e9db3bec53b598c37d5337ee270/docs/contribute/index.qmd#L38

As soon as pixi supports editable installs, this will not be necessary anymore and we can remove `--environment=dev` again.

Steps:

1. Add [editable dependency](https://pixi.sh/latest/reference/configuration/#path) and remove corresponding `install-*` tasks
2. Change all instances of  `pixi run --environment=dev install` to  `pixi run install`

### PR DESCRIPTION
**Pull request by Jingru923**
Fixes #1324 

And reimplement all the changes in PR #1400

### CODE REVIEW

**Comment by Jingru923** [comment]
I have also changed the teamcity configuration. Now in the two test pipeline (linux - Test Ribasim Binaries and windows - Test Ribasim Binaries) have no `pixi run install-python` in the build step

File: .teamcity/Ribasim_Linux/buildTypes/Ribasim_Linux_TestRibasimBinaries.xml [hunk 1]
```diff
@@ -29,7 +29,7 @@ pixi --version]]></param>
 source /usr/share/Modules/init/bash
 
 module load pixi
-pixi run --environment=dev install
+pixi run install
```
Comment (visr) [inline .teamcity/Ribasim_Linux/buildTypes/Ribasim_Linux_TestRibasimBinaries.xml:32]: The `install` task no longer exists, right? And `install-ci`, installs Julia, which isn't needed here. So I guess this line can be removed. Can you also trigger TeamCity builds on this branch to make sure that still works?

File: .teamcity/Ribasim/buildTypes/Ribasim_GenerateTestmodels.xml [hunk 2]
```diff
@@ -19,7 +19,7 @@ source /usr/share/Modules/init/bash
 
 module load pixi
 pixi --version
-pixi run --environment=dev install-ci]]></param>
+pixi run install-ci]]></param>
```
Comment (visr) [inline .teamcity/Ribasim/buildTypes/Ribasim_GenerateTestmodels.xml:22]: I don't think this task is needed anymore.

File: .github/workflows/qgis.yml [hunk 3]
```diff
@@ -27,7 +27,7 @@ jobs:
           pixi-version: "latest"
       - name: Prepare pixi
         run: |
-          pixi run --environment=dev install-ci
+          pixi run install-ci
```
Comment (visr) [inline .github/workflows/qgis.yml:30]: I don't think this task is needed anymore.

File: .github/workflows/python_codegen.yml [hunk 4]
```diff
@@ -20,7 +20,7 @@ jobs:
         with:
           pixi-version: "latest"
       - name: Prepare pixi
-        run: pixi run --environment=dev install-ci
+        run: pixi run install-ci
```
Comment (visr) [inline .github/workflows/python_codegen.yml:23]: I don't think this task is needed anymore.

File: .github/workflows/mypy.yml [hunk 5]
```diff
@@ -20,7 +20,7 @@ jobs:
         with:
           pixi-version: "latest"
       - name: Prepare pixi
-        run: pixi run --environment=dev install-ci
+        run: pixi run install-ci
```
Comment (visr) [inline .github/workflows/mypy.yml:23]: I don't think this task is needed anymore.

File: .teamcity/Ribasim_Windows/buildTypes/Ribasim_Windows_TestRibasimBinaries.xml [hunk 6]
```diff
@@ -21,7 +21,7 @@
       </runner>
       <runner id="RUNNER_1503" name="Run tests" type="simpleRunner">
         <parameters>
-          <param name="script.content"><![CDATA[pixi run --environment=dev install
+          <param name="script.content"><![CDATA[pixi run install
```
Comment (visr) [inline .teamcity/Ribasim_Windows/buildTypes/Ribasim_Windows_TestRibasimBinaries.xml:24]: I don't think this task is needed anymore.

File: docs/contribute/core.qmd [hunk 7]
```diff
@@ -37,7 +37,7 @@ You can let it start automatically by following these [instructions](https://tim
 There is a section on editors and IDEs for Julia on <https://julialang.org/>, scroll down to see it.
 We use and recommend Microsoft's free editor [Visual Studio Code](https://code.visualstudio.com/).
 When combined with the [Julia extension](https://www.julia-vscode.org/) it provides a powerful and interactive development experience.
-Make sure to have the correct environment when opening your IDE by running `open-vscode.bat`, or opening a `pixi --environment=dev shell` and then calling the command to open the editor of your choice.
+Make sure to have the correct environment when opening your IDE by running `open-vscode.bat`, or opening a `pixi shell` and then calling the command to open the editor of your choice.
```
Comment (visr) [inline docs/contribute/core.qmd:40]: Good to check, but I think `pixi --environment=dev shell` is still needed here, just like `open-vscode.bat` also still using `pixi run --environment=dev code .`.

**Review by visr — COMMENTED** [review]
Left some comments. Before this is merged we should manually trigger the TeamCity jobs on this PR, since it is likely to affect it.

File: .teamcity/Ribasim_Linux/buildTypes/Ribasim_Linux_TestRibasimBinaries.xml (hunk 1 above)
Comment (Jingru923) [reply .teamcity/Ribasim_Linux/buildTypes/Ribasim_Linux_TestRibasimBinaries.xml:32]: I removed it and triggered both Linux and Windows Ribasim. They worked.

**Review by SouthEndMusic — APPROVED** [review]
I see no problems with this, because it's mainly deletions. You shouldn't only rely on my judgement for this though

**Pull request merged**